.circleci/config.yml

```diff
@@ -0,0 +1,288 @@
+    description: restore npm package cache
+    parameters:
+      suffix:
+        type: string
+        default: default
+    steps:
+      - restore_cache:
+          keys:
+            - npm-packages-{{ checksum "package-lock.json" }}-<< parameters.suffix >>
+
+  save_build_cache:
+    description: save npm package cache
+    parameters:
+      suffix:
+        type: string
+        default: default
+    steps:
+      - save_cache:
+          key: npm-packages-{{ checksum "package-lock.json" }}-<< parameters.suffix >>
+          paths:
+            - ./node_modules
+
+  run_npm_ci:
+    description: install npm dependencies
+    steps:
+      - run: npm ci
+
+  run_npm_build:
+    description: build project in << parameters.node_env >> mode
+    parameters:
+      node_env:
+        type: enum
+        default: production
+        enum: ["production", "test"]
+    steps:
+      - run:
+          command: npm run build --if-present
+          environment:
+            NODE_ENV: << parameters.node_env >>
+
+  run_functional_tests:
+    description: run functional tests
+    parameters:
+      retry_once:
+        type: enum
+        default: ""
+        enum: ["", "y"]
+      legacy_bundling:
+        type: boolean
+        default: false
+    steps:
+      ## NOTE: by setting the configured python to /bin/false we are forcing
+      ## the production mode tests to fail if any of the dependencies is a
+      ## binary dependency that is build using node-gyp.
+      - run: npm config set python /bin/false
+      - configure_global_npm
+      - run:
+          ## (See #1082 for rationale).
+          name: run functional tests in npm production environment
+          command: npm run test-functional
+          environment:
+            TEST_PRODUCTION_MODE: 1
+            CI_RETRY_ONCE: << parameters.retry_once >>
+      - when:
+          condition: << parameters.legacy_bundling >>
+          steps:
+            - run:
+                ## (See #1639 for rationale).
+                name: run functional tests in npm legacy bundling mode
+                command: npm run test-functional
+                environment:
+                  TEST_PRODUCTION_MODE: 1
+                  TEST_LEGACY_BUNDLING: 1
+                  CI_RETRY_ONCE: << parameters.retry_once >>
+      - run: npm config delete python
+
+  audit_deps:
+    description: audit npm deps
+    steps:
+      - run: npm run audit-deps
+
+  lint_commit_conventions:
+    description: lint commit message conventions
+    steps:
+      - run: npm run github-pr-title-lint
+
+  # This is required to avoid a `EACCES` when running `npm link` (which is
+  # executed in the test suite).
+  configure_global_npm:
+    description: create custom directory for global npm installs
+    steps:
+      # NOTE: keep the following steps compatible with both bash and cmd.exe
+      - run: cd .. && mkdir .npm-global
+      - run: npm config set prefix '../.npm-global'
+
+workflows:
+  node-tests:
```

Comment:
We use `default-workflow` elsewhere because it does more than running test suites.